base/abstractarray.jl

```diff
@@ -1580,8 +1580,8 @@ reduce(::typeof(hcat), A::AbstractVector{<:AbstractVecOrMat}) =
 # helper functions
 cat_size(A) = (1,)
 cat_size(A::AbstractArray) = size(A)
-cat_size(A, d) = 1
-cat_size(A::AbstractArray, d) = size(A, d)
+cat_size(A, d::Int) = 1
+cat_size(A::AbstractArray, d::Int) = size(A, d)
```

Comment:
```suggestion
cat_size(A, d) = 1
cat_size(A::AbstractArray, d) = size(A, d)
```

Maybe preserve the original signatures?  This change seems unnecessary/unrelated, no?  And it'll add ambiguities in, e.g., HDF5:

https://juliahub.com/ui/CodeSearch?q=cat_size&u=all&t=all

Reply:
I can't recall why I did that... Reverting